lib/ansible/executor/play_iterator.py

```diff
@@ -174,7 +174,7 @@ def __init__(self, inventory, play, play_context, variable_manager, all_vars, st
         setup_task = Task(block=setup_block)
         setup_task.action = 'setup'
         setup_task.name = 'Gathering Facts'
-        setup_task.tags = ['always']
+        setup_task.tags = self._play._get_attr_tags()
```

Comment:
This is incorrect, the always tag was added as a 'local' tag to the task specifically to keep the previous behaviour of fact gathering executing irregardless of tags. Even if you wanted to disable this, you should not need to set to the play tags as the task templating itself would run this inheritance so you would set to an empty list or null for that case.